packages/shared/lib/core/layer-2/actions/getLayer2NativeTokens.ts

```diff
@@ -0,0 +1,51 @@
+import { evmAddressToAgentID, getAgentBalanceParameters, getSmartContractHexName, ISC_SANDBOX_ABI } from '@core/layer-2'
+import { network } from '@core/network'
+import { TOKEN_ID_BYTE_LENGTH } from '@core/token'
+import { Converter } from '@iota/util.js'
+import { get } from 'svelte/store'
+import { ISC_MAGIC_CONTRACT_ADDRESS } from '../constants'
+
+// TODO: remove this hardcoded address and use the account's EVM address instead
+const HARDCODED_EVM_ADDRESS = 'WRITE_YOUR_EVM_ADDRESS_HERE'
+
+// TODO: missing set native tokens, which converts the function to getAndSetLayer2NativeTokens
+export async function getLayer2NativeTokens(): Promise<{ amount: bigint; id: string }[]> {
```

Comment:
We should change the name to `getWrappedNativeTokens` then we can see if we can re-use some code for `getNativeL2Tokens` or even `getL2Tokens` which would be a combination of both of those. All of these would be per chain